File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/remote/MediaUrl.kt
package com.visvine.mobile.data.remote

import com.visvine.mobile.core.AppConfig

/**
 * Faithful port of `resolveMediaUrl` in src/services/api.ts — one of the few
 * pieces of real client logic worth porting exactly (covered by MediaUrlTest).
 *
 *  - null / empty            → null
 *  - absolute http(s)/data:  → unchanged
 *  - leading-slash path      → prefixed with the API base
 *  - bare token (no slash)   → left alone
 */
object MediaUrl {
    private val absolute = Regex("^(https?:|data:)")

    fun resolve(url: String?): String? {
        if (url.isNullOrEmpty()) return null
        if (absolute.containsMatchIn(url)) return url
        if (url.startsWith("/")) return "${AppConfig.apiBaseUrl}$url"
        return url
    }
}

File: apps/mobile/ios/Visvine/Networking/MediaURL.swift
import Foundation

/// Faithful port of `resolveMediaUrl` (src/services/api.ts), covered by MediaURLTests.
///  - nil / empty           → nil
///  - absolute http(s)/data → unchanged
///  - leading-slash path    → prefixed with the API base
///  - bare token            → left alone
enum MediaURL {
    private static let absolute = try! NSRegularExpression(pattern: "^(https?:|data:)")

    static func resolve(_ url: String?) -> String? {
        guard let url, !url.isEmpty else { return nil }
        let range = NSRange(url.startIndex..., in: url)
        if absolute.firstMatch(in: url, range: range) != nil { return url }
        if url.hasPrefix("/") { return AppConfig.apiBase + url }
        return url
    }
}
